test/scripts/mutex.js

```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 const should = require('chai').should(); // eslint-disable-line
-const Mutex = require('../../lib/mutex');
+const Mutex = require('../../dist/mutex');
```

Comment:
Maybe we can use `ts-mocha` for unit testing, so no need to import from `dist`.

Reply:
> 
> 
> Maybe we can use `ts-mocha` for unit testing, so no need to import from `dist`.

ts-mocha doesn't seem to be very up to date, I've used mocha with ts-node compilation. This might be better.